src/Uno.UI/Extensions/ViewExtensions.iOSmacOS.cs

```diff
@@ -2,24 +2,43 @@
 
 using System.Collections.Generic;
 using Windows.UI.Xaml;
+using System;
+
+
+#if __IOS__
+using UIKit;
+using View = UIKit.UIView;
+#elif __MACOS__
+using AppKit;
+using View = AppKit.NSView;
+#endif
+
 
 namespace Uno.UI
 {
 	public static partial class ViewExtensions
 	{
 		public static string ShowLocalVisualTree(this UIElement element, int fromHeight = 0)
 		{
-#if __MACOS__
-			return AppKit.UIViewExtensions.ShowLocalVisualTree(element as AppKit.NSView, fromHeight);
-#else
-			return UIKit.UIViewExtensions.ShowLocalVisualTree(element as UIKit.UIView, fromHeight);
-#endif
+			return UIViewExtensions.ShowLocalVisualTree(element as View, fromHeight);
 		}
 
-#if __IOS__
-		internal static IEnumerable<UIKit.UIView> GetChildren(this UIElement element) => element.ChildrenShadow;
-#elif __MACOS__
-		internal static IEnumerable<AppKit.NSView> GetChildren(this UIElement element) => element.ChildrenShadow;
-#endif
+		internal static IEnumerable<View> GetChildren(this UIElement element) => element.ChildrenShadow;
+
+		internal static TResult? FindLastChild<TParam, TResult>(this View group, TParam param, Func<View, TParam, TResult?> selector)
+			where TResult : class
+		{
+			var subviews = group.Subviews;
```

Comment:
Let's use the `IShadowChildrenProvider` instead (accessing `.Subviews` is costly).